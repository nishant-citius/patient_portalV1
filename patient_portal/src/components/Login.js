import { React, useState } from "react";

const Login = () => {
  const tempUser = {
    username: "",
    password: "",
  };
  const [user, setUser] = useState(tempUser);

  return (
    <>
      <div className="container">
        <h4 className="text-center">Login Page</h4>
        <div className="row justify-content-center">
          <div className="col-8"></div>
          <form className="login-form">
            <div className="form-group">
              <label htmlFor="user name">User Name</label>
              <input type="text" className="form-control" name="username" />
              <small id="emailHelp" className="form-text text-muted">
                Forgot username?
              </small>
            </div>

            <div className="form-group">
              <label htmlFor="password">Password</label>
              <input
                type="password"
                className="form-control"
                name="user name"
              />
              <small id="passwordHelp" className="form-text text-muted">
                Forgot password?
              </small>
            </div>
            <br />
            <div>
              <label class="form-check-label" htmlFor="Check1">
                Remember me
              </label>
              <input type="checkbox" className="form-check-input" id="Check1" />
            </div>
            <br />
            <button type="submit" className="btn btn-primary">
              Login
            </button>
          </form>
        </div>
      </div>
    </>
  );
};

export default Login;
