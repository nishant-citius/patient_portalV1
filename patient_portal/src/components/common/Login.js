import { React, useEffect } from "react";
import { useHistory } from "react-router";
import "./common_style.css";
import { connect } from "react-redux";
import * as actionCreator from "../../redux/actions/userActionCreater";
import "../common/common_style.css";
import {
  Grid,
  Paper,
  Avatar,
  TextField,
  Button,
  Typography,
  Link,
} from "@material-ui/core";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import InputAdornment from "@material-ui/core/InputAdornment";
import AccountCircle from "@material-ui/icons/AccountCircle";
import LockIcon from "@material-ui/icons/Lock";
import { Formik, Form, Field, ErrorMessage } from "formik";
import * as Yup from "yup";
import InitializeReduxState from "./InitializeReduxState";

const paperStyle = {
  padding: 20,
  height: "70vh",
  width: 280,
  margin: "60px auto",
  marginTop: "110px",
};
const avatarStyle = { backgroundColor: "#1bbd7e" };
const btnstyle = { margin: "8px 0" };

const Login = (props) => {
  const initialValues = {
    email: "",
    password: "",
  };

  const validationSchema = Yup.object().shape({
    email: Yup.string().email("Please enter valid email").required("Required"),
    password: Yup.string("Enter your password")
      .required("Required")
      .min(4, "Password should be of minimum 4 characters length"),
  });

  const onSubmit = (values) => {
    const payload = { email: values.email, password: values.password };
    props.login(payload);
  };

  let history = useHistory();

  useEffect(() => {
    if (props.isLoggedIn) {
      props.flashNotification({
        message: "Login Succeessful...",
        type: "success",
      });
      if (props.role === "admin") {
        history.push("/admin");
      } else if (props.role === "patient") {
        patientStatus();
      } else {
        history.push("/physician");
      }
    }
  }, [0]);

  const patientStatus = () => {
    if (props.currentUser.isActive) {
      history.push("/demographics");
    } else {
      history.push("/patientinactive");
    }
  };

  return (
    <>
      <Grid>
        <Paper elevation={10} style={paperStyle}>
          <Grid align="center">
            <Avatar style={avatarStyle}>
              <LockOutlinedIcon />
            </Avatar>
            <br />
            <h4>Sign In</h4>
          </Grid>
          <Formik
            initialValues={initialValues}
            onSubmit={onSubmit}
            validationSchema={validationSchema}
          >
            {(props) => (
              <Form>
                <Field
                  as={TextField}
                  label="Email"
                  margin="normal"
                  type="text"
                  name="email"
                  // onChange={handleUserChange}
                  placeholder="Enter email"
                  fullWidth
                  InputProps={{
                    startAdornment: (
                      <InputAdornment position="start">
                        <AccountCircle />
                      </InputAdornment>
                    ),
                  }}
                  variant="standard"
                  helperText={<ErrorMessage name="email" />}
                />
                <Field
                  as={TextField}
                  label="password"
                  placeholder="Enter password"
                  type="password"
                  name="password"
                  // onChange={handleUserChange}
                  fullWidth
                  InputProps={{
                    startAdornment: (
                      <InputAdornment position="start">
                        <LockIcon />
                      </InputAdornment>
                    ),
                  }}
                  helperText={<ErrorMessage name="password" />}
                />

                <Button
                  type="submit"
                  color="primary"
                  variant="contained"
                  style={btnstyle}
                  fullWidth
                >
                  Sign in
                </Button>
              </Form>
            )}
          </Formik>
          <Typography> Do you have an account ?</Typography>
          <a href="/registeruser">Sign Up</a>
          <p className="text text-danger fw-bold text-center">
            {props.globalmessage}
          </p>
        </Paper>
      </Grid>
      <InitializeReduxState />
    </>
  );
};

const mapStatetoProps = (state) => {
  return {
    isLoggedIn: state.login.isLoggedIn,
    role: state.login.role,
    globalmessage: state.login.globalmessage,
    authToken: state.login.authToken,
    currentUser: state.login.loggedUserInfo,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    login: (user) => dispatch(actionCreator.Login(user)),
  };
};

let hof = connect(mapStatetoProps, mapDispatchToProps);
export default hof(Login);
