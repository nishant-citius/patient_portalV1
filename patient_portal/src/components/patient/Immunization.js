import { React, useState, useEffect } from "react";
import { connect } from "react-redux";
import * as actionCreator from "../../redux/actions/userActionCreater";
import { Formik, Form, Field, ErrorMessage, FieldArray } from "formik";
import * as Yup from "yup";
import { useHistory } from "react-router";
import "../admin/admin.css";
import {
  makeStyles,
  Container,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TablePagination,
  TableRow,
} from "mui";

const useStyles = makeStyles((theme) => ({
  root: {
    height: "100vh",
    width: "80vw",
  },
}));

const Immunization = (props) => {
  const classes = useStyles();
  const [isAvailable, setIsAvailable] = useState(false);
  useEffect(() => {
    if (props.isLoggedIn) {
      if (props.immunizationDetails) {
        setIsAvailable(true);
      }
    }
  }, []);
  const initialValues = {
    age_category: "",
    vaccine_brand: "",
    dose_detail: "",
    general_vaccine: [
      {
        vaccine_name: "",
        vaccine_date: "",
      },
    ],
    userid: props.currentUser.id,
  };

  const validationSchema = Yup.object().shape({
    age_category: Yup.string().required("Required"),
    vaccine_brand: Yup.string().required("Required"),
    dose_detail: Yup.string().required("Required"),
    // general_vaccine: Yup.string().required("Required"),
  });

  const onSubmit = (values) => {
    let gv = values.general_vaccine.map((v) => {
      let temp = {};
      temp.vaccine_name = v.vaccine_name;
      temp.vaccine_date = v.vaccine_date;
      return temp;
    });

    const payload = {
      first_name: props.currentUser.fName,
      last_name: props.currentUser.lName,
      age_category: values.age_category,
      vaccine_brand: values.vaccine_brand,
      dose_detail: values.dose_detail,
      general_vaccine: gv,
      id: props.currentUser.id,
    };
    props.immunization(payload);
    props.flashNotification({
      message: "Immunization added...",
      type: "success",
    });
    props.getPatientImmunization(props.currentUser.id);
    history.push("/medic_allergy");
  };

  let history = useHistory();

  // useEffect(() => {
  //   if (props.statusCode === 201) {
  //     history.push("/medic_allergy");
  //   }
  // });
  return (
    <>
      {isAvailable ? (
        <div className="container">
          <div className="card shadow-lg p-10 mb-6 bg-white rounded">
            <div className="card-header text-center">
              <h4 className="text-success">Immunization Details</h4>
            </div>
            <div className="card-body text-center">
              <h5 className="text-primary">Covid Vaccine Details</h5>

              <TableContainer component={Paper} style={{ marginTop: "20px" }}>
                <Table>
                  <TableHead className="tablehead">
                    <TableRow>
                      <TableCell className="tableCell">Age Category</TableCell>
                      <TableCell className="tableCell">Vaccine Brand</TableCell>
                      <TableCell className="tableCell">Dose Details</TableCell>
                    </TableRow>
                  </TableHead>
                  <TableBody>
                    <TableRow>
                      <TableCell>
                        {props.immunizationDetails.age_category}
                      </TableCell>
                      <TableCell>
                        {props.immunizationDetails.vaccine_brand}
                      </TableCell>
                      <TableCell>
                        {props.immunizationDetails.dose_detail}
                      </TableCell>
                    </TableRow>
                  </TableBody>
                </Table>
              </TableContainer>
              <br></br>

              <h5 className="text-primary">General Vaccine Details</h5>
              <TableContainer component={Paper} style={{ marginTop: "20px" }}>
                <Table>
                  <TableHead className="tablehead">
                    <TableRow>
                      <TableCell className="tableCell">Sr.No</TableCell>
                      <TableCell className="tableCell">Vaccine Name</TableCell>
                      <TableCell className="tableCell">Vaccine Date</TableCell>
                    </TableRow>
                  </TableHead>
                  <TableBody>
                    {props.immunizationDetails.general_vaccine.map(function (
                      item,
                      index
                    ) {
                      return (
                        <TableRow key={index}>
                          <TableCell>{index + 1}</TableCell>
                          <TableCell>{item.vaccine_name}</TableCell>
                          <TableCell>{item.vaccine_date}</TableCell>
                        </TableRow>
                      );
                    })}
                  </TableBody>
                </Table>
              </TableContainer>
            </div>
          </div>
        </div>
      ) : (
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
          onSubmit={onSubmit}
          enableReinitialize
        >
          {(props) => (
            <div className="container">
              <div className="card shadow-lg p-10 mb-6 bg-white rounded">
                <div className="card-header">Immunization Details</div>
                <div className="card-body">
                  <Form>
                    <div className="form-group">
                      <label htmlFor="covid_vaccine">
                        COVID-19 Vaccine Details
                      </label>
                      <div className="row">
                        <div className="col-12 col-md-4">
                          <label htmlFor="age_category">Age Category</label>
                          <Field
                            as="select"
                            className="form-control"
                            name="age_category"
                          >
                            <option value="">Select</option>
                            <option value="18_44">18-44</option>
                            <option value="45">45+</option>
                          </Field>
                          <div className="error">
                            <ErrorMessage name="age_category" />
                          </div>
                        </div>
                        <div className="col-12 col-md-4">
                          <label htmlFor="vaccine_brand">Vaccine Brand</label>
                          <Field
                            type="text"
                            className="form-control"
                            name="vaccine_brand"
                            placeholder="Please Enter Covid Vaccine Details"
                          />
                          <div className="error">
                            <ErrorMessage name="vaccine_brand" />
                          </div>
                        </div>
                        <div className="col-12 col-md-4">
                          <label htmlFor="dose_detail">Dose Details</label>
                          <Field
                            as="select"
                            className="form-control"
                            name="dose_detail"
                          >
                            <option value="">Select</option>
                            <option value="1">1</option>
                            <option value="2">2</option>
                          </Field>
                          <div className="error">
                            <ErrorMessage name="dose_detail" />
                          </div>
                        </div>
                      </div>
                    </div>

                    <div className="form-group mt-4">
                      <label htmlFor="general_vaccine">
                        Other general Vaccines
                      </label>
                      <FieldArray name="general_vaccine">
                        {(fieldArrayProps) => {
                          const { push, remove, form } = fieldArrayProps;
                          const { values } = form;
                          const { general_vaccine } = values;
                          return (
                            <div>
                              {general_vaccine.map((general_vaccine, index) => (
                                <div className="row" key={index}>
                                  <div className="col-12 col-md-5">
                                    <label htmlFor="">Vaccine Name</label>
                                    <Field
                                      type="text"
                                      className="form-control"
                                      name={`general_vaccine[${index}].vaccine_name`}
                                    />
                                  </div>
                                  <div className="col-12 col-md-5">
                                    <label htmlFor="">Vaccine Date</label>
                                    <Field
                                      type="date"
                                      className="form-control"
                                      name={`general_vaccine[${index}].vaccine_date`}
                                    />
                                  </div>
                                  <div className="col-12 col-md-2">
                                    {index > 0 && (
                                      <button
                                        className="btn btn-danger btn-number  btn_wdth"
                                        type="button"
                                        onClick={() => remove(index)}
                                      >
                                        -
                                      </button>
                                    )}

                                    <button
                                      className="btn btn-success btn-number mrg_20 btn_wdth"
                                      type="button"
                                      onClick={() => push("")}
                                    >
                                      +
                                    </button>
                                  </div>
                                </div>
                              ))}
                            </div>
                          );
                        }}
                      </FieldArray>
                    </div>
                    <button type="submit" className="btn btn-primary mt-3">
                      Submit
                    </button>
                  </Form>
                </div>
              </div>
            </div>
          )}
        </Formik>
      )}
    </>
  );
};

const mapStateToProps = (state) => {
  return {
    currentUser: state.login.loggedUserInfo,
    immunizationDetails: state.patientImmunization.patientImmunization,
    isLoggedIn: state.login.isLoggedIn,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    immunization: (newuser) =>
      dispatch(actionCreator.AddImmunizationsAsync(newuser)),
    getPatientImmunization: (userId) =>
      dispatch(actionCreator.GetPatientImmunization(userId)),
    updateImmunization: (userId, data) =>
      dispatch(actionCreator.UpdatePatientImmunization(userId, data)),
  };
};

let hof = connect(mapStateToProps, mapDispatchToProps);
export default hof(Immunization);
