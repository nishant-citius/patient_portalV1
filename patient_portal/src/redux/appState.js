let appState = {
  users: [],
  demographics: [],
  immunization: [],
  vitals: [],
  dietplan: [],
  medication_allergies: [],
  isLoggedIn: false,
  authToken: "",
  globalmessage: "",
  role: "",
  loggedUserInfo: {},
  physicians: [],
  patients: [],
  nurses: [],
  immunizations: [],
  userDetails: {},
  profileImage: "",
  inactiveUsers: [],
  approvedAppointments: [],
  patientDemographics: {},
  patientImmunization: {},
  specialisedPhysicians: [],
  appointmentsDetails: [],
  inactiveUserCount: 0,
  approvedAppointmentCount: 0,
  patientMedicationAllergy: [],
  labReports: [],
  pendingAppointmentCount: 0,
  pendingAppointments:[],
};

export default appState;
