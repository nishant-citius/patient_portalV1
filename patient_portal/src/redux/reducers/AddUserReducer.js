import appState from "../appState";
import * as actions from "../actions/userActions";

function AddUserReducer(state = appState, action) {
  if (action.type === actions.ADD_USER) {
    return {
      globalmessage: action.payload.globalmessage,
    };
  }

  return state;
}

export { AddUserReducer };
