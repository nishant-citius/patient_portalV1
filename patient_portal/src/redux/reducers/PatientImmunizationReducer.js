import appState from "../appState";
import * as actions from "../actions/userActions";

export function PatientImmunizationReducer(state = appState, action) {
  if (action.type === actions.GET_PATIENT_IMMUNIZATION) {
    return {
      globalmessage: action.payload.globalmessage,
      patientImmunization: action.payload.userImmunization,
    };
  }

  return state;
}
