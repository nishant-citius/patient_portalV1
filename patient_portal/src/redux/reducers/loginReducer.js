import appState from "../appState";
import * as actions from "../actions/userActions";

function LoginReducer(state = appState , action){
    if (action.type === actions.LOGIN || action.type === actions.LOGOUT) {
      return {
        globalmessage: action.payload.globalmessage,
        isLoggedIn: action.payload.isLoggedIn,
        role: action.payload.role,
        authToken: action.payload.authToken,
        loggedUserInfo: action.payload.loggedUserInfo,
      };
    }

    return state;

}
export {LoginReducer} ;
