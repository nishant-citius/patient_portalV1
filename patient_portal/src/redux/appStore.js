import { createStore, combineReducers } from "redux";
import { applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

import { DemographicsReducer } from "./reducers/Demographicsreducer";
import Immunizationreducer from "./reducers/Immunizationreducer";
import VitalsReducer from "./reducers/Vitalsreducer";
import DietPlanReducer from "./reducers/DietPlanreducer";
import { LoginReducer } from "./reducers/loginReducer";
import { MedicationandAllergiesReducer } from "./reducers/MedicationandAllergiesreducer";
import { RegisterReducer } from "./reducers/RegisterReducer";
import { PhysicianReducer } from "./reducers/userreducers/PhysicianReducer";
import { AddUserReducer } from "./reducers/userreducers/AddUserReducer";
import { GetAllUsersReducer } from "./reducers/userreducers/GetAllUsersReducers";
import { EditUserReducer } from "./reducers/userreducers/EditUserReducer";
import { PatientReducer } from "./reducers/userreducers/PatientReducer";
import { UserDetailsReducer } from "./reducers/userreducers/UserDetailsReducer";
import { updateprofilepicreducer } from "./reducers/updateprofilepicreducer";
import { InactiveUsersReducer } from "./reducers/userreducers/InactiveUsersReducer";
import { PatientDemographicsReducer } from "./reducers/PatientDemographicsReducer";
import { GetNurseReducer } from "./reducers/userreducers/GetNurseReducer";
import { PatientImmunizationReducer } from "./reducers/PatientImmunizationReducer";
import { GetImmunizationReducer } from "./reducers/userreducers/GetImmunizationReducer";
import { SpecialityPhysicianReducer } from "./reducers/userreducers/SpecialityPhysicianReducer";
import { GetAllAppointmentReducer } from "./reducers/GetAllAppointmentReducer";
import { DeleteUserReducer } from "./reducers/userreducers/DeleteUserReducer";
import { GetVitalsReducer } from "./reducers/GetVitalReducer";
import { GetDietPlanReducer } from "./reducers/GetDietPlanReducer";
import { UpdateImmunizationReducer } from "./reducers/UpdateImmunizationReducer";
import { PatientMedicationAllergyReducer } from "./reducers/PatientMedicationAllergyReducer";
import { GetAllDemographicsReducer } from "./reducers/userreducers/GetAllDemographicsReducer";
import { ApprovedAppointmentsReducer } from "./reducers/userreducers/ApprovedAppointmentsReducer";
import { UpdateDemographicsReducer } from "./reducers/UpdateDemographicsReducer";
import { UpdateMedicationAndAllergiesReducer } from "./reducers/updateMedicationAllergyReducer";
import { PendingAppointmentsReducer } from "./reducers/userreducers/pendingAppointmentReducer";

/** combine reducers*/
let rootReducer = combineReducers({
  register: RegisterReducer,
  login: LoginReducer,
  demographics: DemographicsReducer,
  immunization: Immunizationreducer, //addImmunization
  vitals: VitalsReducer,
  dietplan: DietPlanReducer,
  getPatientvitals: GetVitalsReducer,
  getPatientdietplan: GetDietPlanReducer,
  medication_allergies: MedicationandAllergiesReducer,
  physicians: PhysicianReducer,
  adduser: AddUserReducer,
  getallusers: GetAllUsersReducer,
  updateusers: EditUserReducer,
  patients: PatientReducer,
  nurses: GetNurseReducer,
  getalldemographics: GetAllDemographicsReducer,
  immunizations: GetImmunizationReducer, //
  userDetails: UserDetailsReducer,
  updateprofile: updateprofilepicreducer,
  inactiveUsers: InactiveUsersReducer,
  approvedAppointments: ApprovedAppointmentsReducer,
  patientDemographics: PatientDemographicsReducer,
  patientImmunization: PatientImmunizationReducer, //patientImmunization
  specilizedPhysicians: SpecialityPhysicianReducer,
  appointmentsDetails: GetAllAppointmentReducer,
  deleteUser: DeleteUserReducer,
  updateImmunization: UpdateImmunizationReducer,
  updateDemographics: UpdateDemographicsReducer,
  patientMedicationAllergy: PatientMedicationAllergyReducer,
  updateMedicationAndAllergies: UpdateMedicationAndAllergiesReducer,
  pendingAppointments :  PendingAppointmentsReducer
});

/**create store  */
let appStore = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default appStore;
