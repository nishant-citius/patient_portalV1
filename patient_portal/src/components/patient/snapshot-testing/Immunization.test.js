import renderer from "react-test-renderer";
//import Demographics from "../../patient/Demographics";
import Login from "../../common/Login";
import { createStore, combineReducers } from "redux";
import { Provider } from "react-redux";
import { LoginReducer } from "redux/reducers/loginReducer";
import Immunization from "../Immunization";
import { PatientImmunizationReducer } from "redux/reducers/PatientImmunizationReducer";
//import {Immunizationreducer} from "redux/reducers/Immunizationreducer";
describe("Immunization snapshot test", () => {
  let store;
  let jsx;
  let rootReducer;

  beforeEach(() => {
    rootReducer = combineReducers({
      login: LoginReducer,
      //immunization: Immunizationreducer,
      patientImmunization: PatientImmunizationReducer,
    });
    store = createStore(rootReducer);
    jsx = (
      <Provider store={store}>
        <Login />
        <Immunization />
      </Provider>
    );
  });
  it("should render properly", () => {
    const tree = renderer.create(jsx).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
