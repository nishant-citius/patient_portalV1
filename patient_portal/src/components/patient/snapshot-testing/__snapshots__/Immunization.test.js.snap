// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Immunization snapshot test should render properly 1`] = `
Array [
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation10 MuiPaper-rounded"
      style={
        Object {
          "height": "70vh",
          "margin": "60px auto",
          "marginTop": "110px",
          "padding": 20,
          "width": 280,
        }
      }
    >
      <div
        align="center"
        className="MuiGrid-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
          style={
            Object {
              "backgroundColor": "#1bbd7e",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <br />
        <h4>
          Sign In
        </h4>
      </div>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink={true}
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
            onClick={[Function]}
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
              disabled={false}
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter email"
              required={false}
              type="text"
              value=""
            />
          </div>
          <p
            className="MuiFormHelperText-root"
          />
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink={true}
          >
            password
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
            onClick={[Function]}
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
              disabled={false}
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter password"
              required={false}
              type="password"
              value=""
            />
          </div>
          <p
            className="MuiFormHelperText-root"
          />
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "margin": "8px 0",
            }
          }
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Sign in
          </span>
        </button>
      </form>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
         Do you have an account ?
      </p>
      <a
        href="/registeruser"
      >
        Sign Up
      </a>
      <p
        className="text text-danger fw-bold text-center"
      >
        
        !!!
      </p>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="card shadow-lg p-10 mb-6 bg-white rounded"
    >
      <div
        className="card-header"
      >
        Immunization Details
      </div>
      <div
        className="card-body"
      >
        <form
          action="#"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="covid_vaccine"
            >
              COVID-19 Vaccine Details
            </label>
            <div
              className="row"
            >
              <div
                className="col-12 col-md-4"
              >
                <label
                  htmlFor="age_category"
                >
                  Age Category
                </label>
                <select
                  className="form-control"
                  name="age_category"
                  onBlur={[Function]}
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    Select
                  </option>
                  <option
                    value="18_44"
                  >
                    18-44
                  </option>
                  <option
                    value="45"
                  >
                    45+
                  </option>
                </select>
                <div
                  className="error"
                />
              </div>
              <div
                className="col-12 col-md-4"
              >
                <label
                  htmlFor="vaccine_brand"
                >
                  Vaccine Brand
                </label>
                <input
                  className="form-control"
                  name="vaccine_brand"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Please Enter Covid Vaccine Details"
                  type="text"
                  value=""
                />
                <div
                  className="error"
                />
              </div>
              <div
                className="col-12 col-md-4"
              >
                <label
                  htmlFor="dose_detail"
                >
                  Dose Details
                </label>
                <select
                  className="form-control"
                  name="dose_detail"
                  onBlur={[Function]}
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    Select
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                </select>
                <div
                  className="error"
                />
              </div>
            </div>
          </div>
          <div
            className="form-group mt-4"
          >
            <label
              htmlFor="general_vaccine"
            >
              Other general Vaccines
            </label>
            <div>
              <div
                className="row"
              >
                <div
                  className="col-12 col-md-5"
                >
                  <label
                    htmlFor=""
                  >
                    Vaccine Name
                  </label>
                  <input
                    className="form-control"
                    name="general_vaccine[0].vaccine_name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="col-12 col-md-5"
                >
                  <label
                    htmlFor=""
                  >
                    Vaccine Date
                  </label>
                  <input
                    className="form-control"
                    name="general_vaccine[0].vaccine_date"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="date"
                    value=""
                  />
                </div>
                <div
                  className="col-12 col-md-2"
                >
                  <button
                    className="btn btn-success btn-number mrg_20 btn_wdth"
                    onClick={[Function]}
                    type="button"
                  >
                    +
                  </button>
                </div>
              </div>
            </div>
          </div>
          <button
            className="btn btn-primary mt-3"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
